src/ansys/fluent/core/session.py

```diff
@@ -343,8 +306,93 @@ def download(self, file_name: str, local_file_path: Optional[str] = None):
         FileNotFoundError
             If the remote file does not exist.
         """
-        if self.file_service:
-            if self.file_service.file_exist(file_name):
-                self.file_service.download_file(file_name, local_file_path)
+        file_service = self._file_service()
+        if file_service:
+            if file_service.file_exist(file_name):
+                file_service.download_file(file_name, local_file_path)
             else:
                 raise FileNotFoundError("Remote file does not exist.")
+
+    def _wait_for_file(self, file_name: str):
+        """Wait for file to get ready for upload or download.
+
+        Parameters
+        ----------
+        file_name : str
+            Case file name
+        Raises
+        ------
+        FileNotFoundError
+            If a case file does not exist.
+        """
+        file_service = self._file_service()
+        start_time = time.time()
+        max_wait_time = 100
```

Comment:
I think another work around for this is to remove time check and just check for `file_service.file_exist()` until case file gets ready for reading or writing.

Reply:
Yes, we shall get rid of the time limit. The "user" will have to fix any "problems" if the function does not return.

Reply:
only check for `file_service.file_exist()` is not working. We can use `sys.maxsize` instead of `max_wait_time = 100`.

![image](https://github.com/ansys/pyfluent/assets/106588300/bf328f92-181a-4c93-8ad0-3e4109b7f8ec)